util/chunk/chunk.go

```diff
@@ -32,45 +32,56 @@ type Chunk struct {
 	columns []*column
 }
 
-// AddFixedLenColumn adds a fixed length column with elemLen and initial data capacity.
-func (c *Chunk) AddFixedLenColumn(elemLen, initCap int) {
+const initialCapacity = 32
+
+// NewChunk creates a new chunk with field types.
+func NewChunk(fields []*types.FieldType) *Chunk {
```

Comment:
Why not add a max row size, so every `Chunk`'s row size is limited to that value, this can help us control our memory usage ?

Reply:
The size is controlled by the executor who writes it.
We don't need to enforce the limit on *Chunk itself.